### PR TITLE
GLM5.2 DSA Indexer Fallback

### PR DESCRIPTION
**Pull request by wenscarl**
## Motivation

  `SGLANG_FP8_PAGED_MQA_LOGITS_TORCH=1` is meant to be the escape hatch that runs the
  FP8 MQA-logits indexer without DeepGEMM. Today it only covers the **dsv4** indexer
  path (`dsv4/metadata.py`, `dsv4/indexer.py`).

  GLM 5.2 (`GlmMoeDsaForCausalLM`, derived from `DeepseekV2ForCausalLM`) runs the
  **dsa** indexer instead, which calls `deep_gemm.get_paged_mqa_logits_metadata` /
  `fp8_paged_mqa_logits` / `fp8_mqa_logits` unconditionally. The env var is therefore
  a no-op for GLM 5.2, and DeepGEMM stays a hard requirement.

  This PR makes the flag effective on the dsa path so GLM 5.2 can be functionally
  validated on hosts where DeepGEMM is unavailable or where the DeepGEMM MQA-logits
  kernels are not usable.

  ## Modifications

  **1. Route the dsa indexer's DeepGEMM entry points through resolvers**
  (`dsa/utils.py`, consumed by `dsa_backend.py` and `dsa/dsa_indexer.py`):

  | resolver | `SGLANG_FP8_PAGED_MQA_LOGITS_TORCH=1` | default |
  |---|---|---|
  | `resolve_paged_mqa_logits_metadata_fn()` | `sglang.kernels.ops.attention.dsv4.get_paged_mqa_logits_metadata` (JIT) | `deep_gemm.get_paged_mqa_logits_metadata` |
  | `resolve_fp8_paged_mqa_logits_fn()` | `_fp8_paged_mqa_logits_torch` → existing `dsv4.indexer.fp8_paged_mqa_logits_torch` | `deep_gemm.fp8_paged_mqa_logits` |
  | `resolve_fp8_mqa_logits_fn()` | `_fp8_mqa_logits_torch` (new) | `deep_gemm.fp8_mqa_logits` |

  The paged and metadata variants reuse existing kernels; only the prefill path needed
  a new pure-torch implementation. It accumulates per-head `relu(q·k)` weighted by the
  gate and applies the fp8 scale, iterating over heads to keep the `[num_q, num_kv]`
  intermediate small. Out-of-range positions are filled with `0.0`, matching the
  already-validated `dsv4.indexer.fp8_paged_mqa_logits_torch` reference. All four call
  sites pass `clean_logits=False`, which is the only mode implemented (asserted).

  **2. Remove the last hard DeepGEMM dependency on this path.** `dsa_backend.py`
  imported `deep_gemm` at module scope solely for `get_num_sms()`, so the module could
  not even be loaded without DeepGEMM installed. Added `resolve_num_sms()`, which reads
  `torch.cuda.get_device_properties(...).multi_processor_count` on the fallback path,
  and dropped the import. Both consumers only use the value to shape the
  schedule-metadata buffer (`new_empty(num_sm + 1, 2)`), so the raw device SM count is
  an acceptable stand-in.

  `configure_deep_gemm_num_sms` (`dsa_indexer.py:488/575`) needs no change: it yields
  immediately when `not ENABLE_JIT_DEEPGEMM`, and that flag is already `False` when the
  DeepGEMM import fails.

  No behavior change when the env var is unset.

  ## Accuracy Tests

  Resolver wiring verified in both directions:

  - **Env var set, DeepGEMM blocked from the import system** (a `sys.meta_path` finder
    raising `ImportError` for `deep_gemm*`, simulating a host without it): all three dsa
    modules import successfully and every resolver resolves — `resolve_num_sms()` → 152,
    metadata → JIT kernel, paged/prefill → the torch implementations.
  - **Env var unset:** all resolvers return the `deep_gemm` callables and
    `resolve_num_sms()` matches `deep_gemm.get_num_sms()` (152 == 152).

  

  ## Speed Tests and Profiling

  Not applicable — this is a correctness/portability fallback, not a performance path.
  `_fp8_mqa_logits_torch` loops over heads in Python with `torch.mm`, and
  `_pad_heads_for_deep_gemm` pads to 32 heads when the model has fewer, so several of
  those matmuls are on zero-weight padding. The default DeepGEMM path is unchanged.

  ## Checklist

  - [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
  - [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
  - [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
  - [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the
  speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
  - [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #33360456827](https://github.com/sgl-project/sglang/actions/runs/33360456827)
Latest PR Test (Extra): :x: [Run #33360456519](https://github.com/sgl-project/sglang/actions/runs/33360456519)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33360456788](https://github.com/sgl-project/sglang/actions/runs/33360456788)

### CODE REVIEW

**Comment by nvpohanh** [comment]
@wenscarl could you fix the conflicts?

**Comment by nvpohanh** [comment]
cc @nvjullin

**Comment by nvpohanh** [comment]
```
✗ FAILED: /actions-runner/_work/sglang/sglang/test/registered/kernels/ops/attention/test_dsa_indexer.py returned exit code 1
```
https://github.com/sgl-project/sglang/actions/runs/31630938791/job/94361342938?pr=32947

@wenscarl could you fix this? thanks

**Comment by nvpohanh** [comment]
Seems to be real failure: https://github.com/sgl-project/sglang/actions/runs/33360456827/job/99394282237?pr=32947
@wenscarl could you fix it?
